opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -78,12 +78,14 @@
    * @param format the Format of the carrier
    * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
    *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
-   * @returns the SpanContext instance extracted from the carrier
+   *
+   * @return the SpanBuilder instance holding context to create a Span.
+   * A noop builder is returned if the carrier held no trace information.
    *
    * @see io.opentracing.propagation.Format
    * @see io.opentracing.propagation.Format.Builtin
    */
-  <C> SpanContext extract(Format<C> format, C carrier);
+  <C> SpanBuilder extract(Format<C> format, C carrier);
```

Comment:
I agree with @bensigelman. Extracting `SpanContext` from a serialized representation does not have to be mixed with span creation. I actually have [use cases](https://github.com/uber/tchannel-go/blob/dev/tracing.go#L227) where doing so is in fact harmful (well, at least wasteful), because I only need the SpanContext. The link is to Go example, but I'll have the exact same pattern in Java.